### DISCUSSION TITLE
Runtime Error with analog input board

### DISCUSSION THREAD

**Discussion by ablot** [harp-tech/protocol#104] [Q&A]
Hi all,
I downloaded the example bonsai workflow from [here](https://github.com/harp-tech/device.analoginput/tree/main/Assets
), but it seems to error with my board. 
I wonder if it's a firmware issue. The "detect devices" tab of the behaviour GUI says that my analog input board as Firmware 1.2, harp version 1.12, hardware version 1.0.
In bonsai the harp package was installed today and is version 3.5.

Should I update the firmware/downgrade the harp package or is the error something else?

Here is the error message:
![error](https://github.com/user-attachments/assets/3d495159-f2f3-4d80-9527-ce0daa74d9ca)

Thanks!

**Comment by bruno-f-cruz** [comment]
Hi! Are you sure you are using a harp analog board and not an harp behavior? Would you mind installing this tool and reporting the output of dotnet harp.toolkit --port COMX?

Also, that workflow seems partly deprecated, can you add the bonsai.config file of your bonsai environment too?
Thanks

**Comment by ablot** [comment]
It is for sure an analog input board. I even got to see the 4 analog signals displayed on bonsai for a bit before it crashed.

That's the output of the toolkit

```
PS C:\Users\Stimulus\Downloads>  dotnet harp.toolkit --port COM4
Harp device found in COM4
DeviceName: AnalogInput
WhoAmI: 1236
Hw: 1.0
Fw: 1.2
Timestamp (s): 1373
```

Sorry, I assumed the workflow would be up-to-date on the repo but it might just be more a development tool than intended for users. Do you know if you happen to have up to date example workflow for each board? (behaviour, analoginput and timestampgenerator in particular) I can build them with the minimum I need but your workflow are a great alternative to user manuals (for instance to understand how to start the analoginput device :) )

Finally the workflow and the bonsai config:
[HardwareUnitTests.zip](https://github.com/user-attachments/files/17255869/HardwareUnitTests.zip)
